arcgis/arcgis.py

```diff
@@ -130,19 +130,31 @@ def esri_to_geojson(self, obj, geom_parser):
             "geometry": geom_parser(obj.get('geometry'))
         }
 
-    def get_json(self, layer, where="1 = 1", fields=[], count_only=False, srid='4326'):
+    def get_json(self, layer, where="1 = 1", fields=[], count_only=False, srid='4326',
+                 input_geom_type=None, input_geom=None, input_srid=None, spatial_rel=None):
         """
         Gets the JSON file from ArcGIS
         """
         params = {
-                'where': where,
-                'outFields': ", ".join(fields),
-                'returnGeometry': True,
-                'outSR': srid,
-                'f': "pjson",
-                'orderByFields': self.object_id_field,
-                'returnCountOnly': count_only
-            }
+            'where': where,
+            'outFields': ", ".join(fields),
+            'returnGeometry': True,
+            'outSR': srid,
+            'f': "pjson",
+            'orderByFields': self.object_id_field,
+            'returnCountOnly': count_only
+        }
+
+        geom_filter_params = {
+            'inSR': input_srid,
+            'geometryType': input_geom_type,
+            'geometry': input_geom,
+            'spatialRel': spatial_rel
+        }
+
+        if all([input_srid, input_geom_type, input_geom]):
```

Comment:
It might make sense to throw a warning here if only some of the four inputs are provided.